cmake/toolchains/mips.cmake

```
@@ -35,6 +35,12 @@ set(THREADS_PTHREAD_ARG "0" CACHE STRING "Result from TRY_RUN" FORCE)
 set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
 set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
 
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+    set(_tbb_mips_pie -pie)
+endif()
+
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -EL -mabi=64 -march=mips64r6 -mcrc -mfp64 -mmt -mtune=mips64r6 -ggdb -ffp-contract=off -mhard-float" CACHE INTERNAL "")
 set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -mvirt -mxpa" CACHE INTERNAL "")
-set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -mvirt -mxpa" CACHE INTERNAL "")  # for tests
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -mvirt -mxpa ${_tbb_mips_pie}" CACHE INTERNAL "")  # for tests
```

```suggestion
set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -mvirt -mxpa $<$<CONFIG:Debug>:-pie>" CACHE INTERNAL "")  # for tests
```

Is it possible to use suggestion above instead of using temp variable? Or is there  some problem with this approach?

---

I have tried this approach but CMake just inserts this generator expression to the command line as link option...

---

```suggestion
set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS}" -mvirt -mxpa $<$<CONFIG:Debug>:-pie> CACHE INTERNAL "")  # for tests
```
How about that?

---

Still doesn't work. Fails on configuration stage with:
```
mips-img-linux-gnu-g++: error: ;-mvirt;-mxpa;$<$<CONFIG:Debug>:-pie>: No such file or directory
```